<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>RTCGA by MarcinKosinski</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='http://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">RTCGA</h1>
      <h2 class="project-tagline">RTCGA package offers download and integration of the variety and volume of TCGA data using patient barcode key, what enables easier data possession.</h2>
      <a href="https://github.com/MarcinKosinski/RTCGA" class="btn">View on GitHub</a>
      <a href="https://github.com/MarcinKosinski/RTCGA/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/MarcinKosinski/RTCGA/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h1>
<a id="mission" class="anchor" href="#mission" aria-hidden="true"><span class="octicon octicon-link"></span></a>Mission</h1>

<p>The Cancer Genome Atlas (TCGA) Data Portal provides a platform for researchers to search, download, and analyze data sets generated by TCGA. It contains clinical information, genomic characterization data, and high level sequence analysis of the tumor genomes[1]. The key is to understand genomics to improve cancer care. </p>

<p>RTCGA package offers download and integration of the variety and volume of TCGA data using patient barcode key, what enables easier data possession. This may have an benefcial infuence on impact on development of science and improvement of patients' treatment. RTCGA is an open-source R package, available to download from Bioconductor[2]. Furthermore, RTCGA package transforms TCGA data to form which is convenient to use in R statistical package. Those data transformations can be a part of statistical analysis pipeline which can be more reproducible with RTCGA.</p>

<h1>
<a id="rtcga" class="anchor" href="#rtcga" aria-hidden="true"><span class="octicon octicon-link"></span></a>RTCGA</h1>

<p>Project is (will be) supported by <a href="https://travis-ci.org/">Travis CI</a> and <a href="https://waffle.io/">waffle.io</a>.</p>

<h5>
<a id="-installation-of-the-rtcga-package-" class="anchor" href="#-installation-of-the-rtcga-package-" aria-hidden="true"><span class="octicon octicon-link"></span></a> Installation of the RTCGA package: </h5>

<p>To get started, install the latest version of <strong>RTCGA</strong> from Bioconductor:</p>

<div class="highlight highlight-Ruby"><pre><span class="pl-c"># not there yet</span></pre></div>

<p>or use:</p>

<div class="highlight highlight-Ruby"><pre><span class="pl-k">if</span> (!<span class="pl-k">require</span>(devtools)) {
    install.packages(<span class="pl-s"><span class="pl-pds">"</span>devtools<span class="pl-pds">"</span></span>)
    <span class="pl-k">require</span>(devtools)
}
install_github(<span class="pl-s"><span class="pl-pds">"</span>MarcinKosinski/RTCGA<span class="pl-pds">"</span></span>)</pre></div>

<p>Make sure you have <a href="http://cran.r-project.org/bin/windows/Rtools/">rtools</a> installed on your computer.</p>

<h5>
<a id="-the-list-of-available-functions-" class="anchor" href="#-the-list-of-available-functions-" aria-hidden="true"><span class="octicon octicon-link"></span></a> The list of available functions: </h5>

<div class="highlight highlight-Ruby"><pre>help(package<span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>RTCGA<span class="pl-pds">"</span></span>)</pre></div>

<p><strong>To run examples below use those packages</strong></p>

<div class="highlight highlight-Ruby"><pre>library(<span class="pl-c1">RTCGA</span>)
library(dplyr)</pre></div>

<h5>
<a id="-information-about-available-tcga-data" class="anchor" href="#-information-about-available-tcga-data" aria-hidden="true"><span class="octicon octicon-link"></span></a> Information about available TCGA data:</h5>

<div class="highlight highlight-Ruby"><pre>infoTCGA()</pre></div>

<h5>
<a id="-cohorts-names" class="anchor" href="#-cohorts-names" aria-hidden="true"><span class="octicon octicon-link"></span></a> Cohorts names:</h5>

<div class="highlight highlight-Ruby"><pre>(cohorts <span class="pl-k">&lt;</span><span class="pl-k">-</span> infoTCGA() <span class="pl-s"><span class="pl-pds">%&gt;</span>% </span>
<span class="pl-s">   names() %<span class="pl-pds">&gt;</span></span><span class="pl-k">%</span> 
   sub(<span class="pl-s"><span class="pl-pds">"</span>-counts<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>, x<span class="pl-k">=</span>.))</pre></div>

<p>Cohorts codes are explained here: <a href="http://gdac.broadinstitute.org/">http://gdac.broadinstitute.org/</a>.</p>

<h5>
<a id="-dates-of-reales-" class="anchor" href="#-dates-of-reales-" aria-hidden="true"><span class="octicon octicon-link"></span></a> Dates of reales: </h5>

<div class="highlight highlight-Ruby"><pre>availableDates()</pre></div>

<p>It is always save to use almost the last date, because sometimes data from the newest date are not available in 100%. So that's why I suggest using such date:</p>

<div class="highlight highlight-Ruby"><pre>date <span class="pl-k">&lt;</span><span class="pl-k">-</span> tail( availableDates(), <span class="pl-c1">2</span> )[<span class="pl-c1">1</span>]</pre></div>

<h3>
<a id="-clinical-data" class="anchor" href="#-clinical-data" aria-hidden="true"><span class="octicon octicon-link"></span></a> Clinical data</h3>

<p>Can be downloaded using default settings. By default the dataset is downloaded from the newest available date of TCGA releases.</p>

<div class="highlight highlight-Ruby"><pre><span class="pl-c">#dir.create( "data" )</span>
downloadTCGA( cancerTypes <span class="pl-k">=</span> cohorts, destDir <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>data/<span class="pl-pds">"</span></span>, date <span class="pl-k">=</span> date )</pre></div>

<h3>
<a id="-mutations-data" class="anchor" href="#-mutations-data" aria-hidden="true"><span class="octicon octicon-link"></span></a> Mutations data</h3>

<p>Let us check if there are Mutations data available for all canser types.</p>

<div class="highlight highlight-Ruby"><pre>checkDataSetsAvailability( cohorts, 
                           pattern <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Mutation_Packager_Calls.Level<span class="pl-pds">"</span></span>, 
                           date <span class="pl-k">=</span> date) <span class="pl-s"><span class="pl-pds">%&gt;</span>%</span>
<span class="pl-s">   unlist() %<span class="pl-pds">&gt;</span></span><span class="pl-k">%</span> 
   cat( sep<span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)</pre></div>

<p>Let us download them.</p>

<div class="highlight highlight-Ruby"><pre>sapply( cohorts, function(element){
   tryCatch({
      downloadTCGA( cancerTypes <span class="pl-k">=</span> element, 
                    dataSet <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Mutation_Packager_Calls.Level<span class="pl-pds">"</span></span>,
                    destDir <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>data/<span class="pl-pds">"</span></span>, 
                    date <span class="pl-k">=</span> date )},
      error <span class="pl-k">=</span> function(cond){
         cat(<span class="pl-s"><span class="pl-pds">"</span>Error: Maybe there weren't mutations data for <span class="pl-pds">"</span></span>, element, <span class="pl-s"><span class="pl-pds">"</span> cancer.<span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
   })
})</pre></div>

<h3>
<a id="-genes-expressions---rna-seq-data-" class="anchor" href="#-genes-expressions---rna-seq-data-" aria-hidden="true"><span class="octicon octicon-link"></span></a> Genes expressions - RNA-Seq data </h3>

<p>Let us check if there are RNA-Seq data available for all canser types.</p>

<div class="highlight highlight-Ruby"><pre>checkDataSetsAvailability( cohorts,
                        <span class="pl-s"><span class="pl-pds">"</span>rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level<span class="pl-pds">"</span></span>,
                        date <span class="pl-k">=</span> date ) <span class="pl-s"><span class="pl-pds">%&gt;</span>%</span>
<span class="pl-s">   unlist() %<span class="pl-pds">&gt;</span></span><span class="pl-k">%</span>  
   cat( sep<span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)</pre></div>

<p>Let us download them.</p>

<div class="highlight highlight-Ruby"><pre>sapply( cohorts, function(element){
   tryCatch({
      downloadTCGA( cancerTypes <span class="pl-k">=</span> element, 
                    dataSet <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level<span class="pl-pds">"</span></span>,
                    destDir <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>data/<span class="pl-pds">"</span></span>, 
                    date <span class="pl-k">=</span> date )},
      error <span class="pl-k">=</span> function(cond){
         cat(<span class="pl-s"><span class="pl-pds">"</span>Error: Maybe there weren't rnaseq data for <span class="pl-pds">"</span></span>, element, <span class="pl-s"><span class="pl-pds">"</span> cancer.<span class="pl-cce">\n</span><span class="pl-pds">"</span></span>)
   })
}) </pre></div>

<h3>
<a id="-untarring-data-" class="anchor" href="#-untarring-data-" aria-hidden="true"><span class="octicon octicon-link"></span></a> Untarring data </h3>

<p>Letâ€™s use <code>untar()</code> function to untar all downloaded sets.</p>

<div class="highlight highlight-Ruby"><pre>list.files( <span class="pl-s"><span class="pl-pds">"</span>data/<span class="pl-pds">"</span></span>) <span class="pl-s"><span class="pl-pds">%&gt;</span>% </span>
<span class="pl-s">   paste0( "data/", .) %<span class="pl-pds">&gt;</span></span><span class="pl-k">%</span>
   sapply( untar, exdir <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>data/<span class="pl-pds">"</span></span> )</pre></div>

<h3>
<a id="-removing-no-longer-needed-targz-files-" class="anchor" href="#-removing-no-longer-needed-targz-files-" aria-hidden="true"><span class="octicon octicon-link"></span></a> Removing no longer needed `tar.gz` files </h3>

<p>After datasets are untarred, the <code>tar.gz</code> files ar no longer needed and can be deleted.</p>

<div class="highlight highlight-Ruby"><pre>list.files( <span class="pl-s"><span class="pl-pds">"</span>data/<span class="pl-pds">"</span></span>) <span class="pl-s"><span class="pl-pds">%&gt;</span>% </span>
<span class="pl-s">   paste0( "data/", .) %<span class="pl-pds">&gt;</span></span><span class="pl-k">%</span>
   grep( pattern <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>tar.gz<span class="pl-pds">"</span></span>, x <span class="pl-k">=</span> ., value <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>) <span class="pl-s"><span class="pl-pds">%&gt;</span>%</span>
<span class="pl-s">   sapply( file.remove )</span></pre></div>

<h4>
<a id="-the-list-of-use-cases-" class="anchor" href="#-the-list-of-use-cases-" aria-hidden="true"><span class="octicon octicon-link"></span></a> The list of use-cases: </h4>

<p>TODO.</p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/MarcinKosinski/RTCGA">RTCGA</a> is maintained by <a href="https://github.com/MarcinKosinski">MarcinKosinski</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>

